app/controllers/events_controller.rb

```diff
@@ -11,17 +9,19 @@ def all
   end
 
   def create
-    event = Apple.new.fire({ message: params[:message] })
-    publish event
 
-    event = Orange.new.receive event
-    publish event
+    originating_data = { message: params[:message] }
+
+    steps = [{ name: 'Apple',  type: 'EventHandler' },
+             { name: 'Orange', type: 'EventHandler' },
+             { name: 'Banana', type: 'EventHandler' },
+             { name: 'Pear',   type: 'EventHandler' }]
 
-    event = Banana.new.receive event
-    publish event
+    steps.each_with_index { |x, i| x[:method] = (i == 0 ? :fire : :receive) }
```

Comment:
I like this line being separate, just so we see "first step is different" concept isolated.